src/widgetastic_patternfly/__init__.py

```diff
@@ -1849,3 +1849,177 @@ def click_location(self, name, handle_alert=True):
+                strptime_format='%m/%d/%Y', locator=None, logger=None): # noqa
+        View.__init__(self, parent=parent, logger=logger)
+
+        self.strptime_format = strptime_format
+        base_locator = './/*[(self::input or self::textarea) and @{}={}]'
+
+        if id:
+            self.locator = base_locator.format('id', quote(id))
+        elif name:
+            self.locator = base_locator.format('name', quote(name))
+        elif locator:
+            self.locator = locator
+        else:
+            raise TypeError('You need to specify either, id, name or locator for DatePicker')
+
+    class HeaderView(View):
+        prev_button = Text(".//*[contains(@class, 'prev')]")
+        next_button = Text(".//*[contains(@class, 'next')]")
+        datepicker_switch = Text(".//*[contains(@class, 'datepicker-switch')]")
+        _elements = {}
+
+        def select(self, value):
+            for el, web_el in self._elements.items():
+                if el == value:
+                    web_el.click()
+                    return True
+
+        @property
+        def active(self):
+            for el, web_el in self._elements.items():
+                if bool(self.browser.classes(web_el) & {'active', 'focused'}):
+                    return el
+
+    @View.nested
+    class date_pick(HeaderView):    # noqa
+        DATES = ".//*[contains(@class, 'datepicker-days')]/table/tbody/tr/td"
+
+        @property
+        def _elements(self):
+            dates = {}
+            for el in self.browser.elements(self.DATES):
+                if not bool({'old', 'new', 'disabled'} & self.browser.classes(el)):
+                    dates.update({int(el.text): el})
+            return dates
+
+    @View.nested
+    class month_pick(HeaderView):   # noqa
+        MONTHS = ".//*[contains(@class, 'datepicker-months')]/table/tbody/tr/td/*"
+
+        @property
+        def _elements(self):
+            months = {}
+            for el in self.browser.elements(self.MONTHS):
+                if not bool({'disabled'} & self.browser.classes(el)):
+                    months.update({el.text: el})
+            return months
+
+    @View.nested
+    class year_pick(HeaderView):    # noqa
+        YEARS = ".//*[contains(@class, 'datepicker-years')]/table/tbody/tr/td/*"
+
+        @property
+        def _elements(self):
+            years = {}
+            for el in self.browser.elements(self.YEARS):
+                if not bool({'old', 'new', 'disabled'} & self.browser.classes(el)):
+                    years.update({int(el.text): el})
+            return years
+
+        def _pick(self, value):
+            for el, web_el in self._elements.items():
+                if el == value:
+                    web_el.click()
+                    return True
+
+        def select(self, value):
+            start_yr, end_yr = [int(item) for item in self.datepicker_switch.read().split('-')]
+            if value > end_yr:
+                for _yr in range(end_yr, value, 10):
```

Comment:
are you sure about amount of transitions ? I haven't checked but it seems it makes one more transition than it should.
as for underscore variables, I meant using just underscore instead of variable name.
you can read more [here](https://hackernoon.com/understanding-the-underscore-of-python-309d1a029edc) about this syntax convention.

Reply:
No transition is fine. I tested it up to a selection of `1900` to `3000`